bin/recog_verify

```diff
@@ -48,15 +55,45 @@ if ARGV.empty?
   exit(1)
 end
 
+def validate(xmlfile_path, xsd_path)
+  schema = Nokogiri::XML::Schema(File.read(xsd_path))
+  doc = Nokogiri::XML(File.read(xmlfile_path))
+  schema.validate(doc)
+end
+
 warnings = 0
 failures = 0
+formatter = Recog::Formatter.new(options, $stdout)
 ARGV.each do |arg|
   Dir.glob(arg).each do |file|
-    ndb = Recog::DB.new(file)
-    verifier = Recog::VerifierFactory.build(options, ndb)
-    verified = verifier.verify
-    failures += verifier.reporter.failure_count
-    warnings += verifier.reporter.warning_count
+    # Create a new reporter per XML file to hold context on success/warn/fails
+    reporter  = Recog::VerifyReporter.new(options, formatter, file)
+
+    begin
+      # Validate the XML database against the recog schema first, if requested
+      if options.schema_file
```

Comment:
This block will validate the XML schema and dump out errors prior to running `<example>` tests. If there are XML errors, the example tests will never run (as the XML is considered malformed anyway). Thus, it's possible to run this, get XML errors in the verify output log, then run it again and get errors with your fingerprints if you violate rules, like missing an example.

Also, we should require the example element at least once via the XSD file once all fingerprint files have them. That way those checks can be done simply through the XML schema rather than having recog_verify manually validate that examples are present.